modules/http-api/src/main/java/org/mule/service/http/api/server/HttpServerFactory.java

```diff
@@ -16,9 +18,20 @@
 public interface HttpServerFactory {
 
   /**
+   * Creates a new {@link HttpServer}.
+   *
    * @param configuration a {@link HttpServerConfiguration} specifying the desired server.
    * @return a newly built {@link HttpServer} based on the {@code configuration}.
    * @throws ConnectionException if the server cannot be created based on the configuration.
    */
   HttpServer create(HttpServerConfiguration configuration) throws ConnectionException;
+
+  /**
+   * Allows to retrieve a previously created {@link HttpServer}, if used from the same context. Keep in mind lifecycle changes to
+   * the retrieved instance won't take effect since only the owner of the server can modify it's status.
+   *
+   * @param name the name the desired {@link HttpServer} was given when created (see {@link HttpServerConfiguration#getName()})
+   * @return an {@link Optional} with the server or an empty one, if none was found.
+   */
+  Optional<HttpServer> lookup(String name);
```

Comment:
Would it make more sense to directly fail if no server with the given name is found?